05_test.sh

```diff
@@ -139,11 +151,22 @@ check_bmh_status(){
 #
 get_vm_ip(){
   local BMH_NAME="${1}"
-  sudo virsh net-dhcp-leases baremetal | grep "${BMH_NAME}" | awk '{print $5}' \
```

Comment:
I just realized that these two IPs aren't the same - previously we'd get the IP for the baremetal network, e.g the one which is used *after* provisioning for the deployed OS.

The changes below switch it to get the provisioning IP, which is potentially only up during provisioning?

I just wanted to ensure that is what you expect, as we do things like `ssh_to_machine` which I think should target the host OS not e.g the deploy ramdisk where it will fail?

Reply:
yes, switching to the provisioning network is done on purpose. The provisioning interface is the primary interface, so it will be configured by default by cloud-init (except if some specific cloud-init is given). The problem with the baremetal network is that you need to run a specific cloud-init to get it configured and it depends on the image you deploy with. This makes it more generic, as it does not rely on what is in the cloud-init. If we would want to check that the cloud-init ran properly, we can add tests over ssh.
About the deploy ramdisk, the credentials will be incorrect, so it will fail to ssh there.

Reply:
@maelk thanks for confirming - sounds reasonable but do we know why the baremetal nic is not coming up, despite the dnsmasq configuration having a reservation for it, is it just because of the nic ordering?

My concern is that we only create dhcp reservations in the libvirt dnsmasq for the baremetal network:

https://github.com/metal3-io/metal3-dev-env/blob/master/vm-setup/roles/common/tasks/main.yml#L33

So if we switch to the provisioning network by default then we require the ironic dnsmasq to allocate and IP, and IIRC that is not a persistent allocation we can necessarily rely on?

I guess for testing it should be fine, but I know from testing on OpenShift that often you would normally expect to deploy via a dedicated provisioning nic, then have an "external" or controlplane nic come up on booting the OS, and at this point there's no access via the provisioning nic.

Reply:
As a side note, @dtantsur mentioned that Ironic does have support for network_data metadata so perhaps in future we can control the desired nic for cloud-init to configure via metadata when https://github.com/metal3-io/baremetal-operator/issues/329 is implemented.

https://specs.openstack.org/openstack/nova-specs/specs/liberty/implemented/metadata-service-network-info.html

Not looked into cloud-init support for that data yet though.